dev/release/verify-release-candidate.sh

```diff
@@ -492,7 +492,7 @@ test_integration() {
   export ARROW_JAVA_INTEGRATION_JAR=$JAVA_DIR/tools/target/arrow-tools-$VERSION-jar-with-dependencies.jar
   export ARROW_CPP_EXE_PATH=$CPP_BUILD_DIR/release
 
-  pip3 install -e dev/archery
+  pip install -e dev/archery
```

Comment:
If you are sure you're using Python 3, I would say `python3 -m pip install ...` is the surest way.

Reply:
This step is invoked from within a conda environment that is set up in the script. I guess recently (January 2020-related?) the pip3 alias disappeared

Reply:
The conda environment wouldn't have been installed in certain flag configuration: 

```bash
TEST_DEFAULT=0 TEST_INTEGRATION_GO=1 dev/release/verify-release-candidate.sh source 0.16.0 2
```

Reply:
Changed the script to **always** use miniconda.